### PR TITLE
Add the registry into KnativeEventingSpec to support custom image registry

### LINKED ISSUE #83: CRD options for image registry and namespace
**Problem**

The images the operator lays down could be from a different image registry hub. (for security and scanning reasons).   Therefore need a way in the CRD to specify the image registry 

Something simple like:   `hub:  myhub.io`

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)**
Which persona is this feature for?
Event consumer (developer)

**Exit Criteria**

The image regs in the yaml manifest generated by the operator has the custom image registry hub.

**Time Estimate (optional):**
How many developer-days do you think this may take to resolve? 1

**Additional context (optional)**
Add any other context about the feature request here.

### PR DESCRIPTION
**Pull request by houshengbo**
# Issue to be fixed

Fixes #83 

## Proposed Changes

* This PR adds the support to download images from private registries for all the deployments.

## Release Note

```release-note
```

### CODE REVIEW

File: pkg/reconciler/knativeeventing/common/images.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline pkg/reconciler/knativeeventing/common/images.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/knativeeventing/common/images_test.go [hunk 2]
```diff
@@ -0,0 +1,253 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline pkg/reconciler/knativeeventing/common/images_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/knativeeventing/common/images.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
```
Comment (mattmoor) [inline pkg/reconciler/knativeeventing/common/images.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/knativeeventing/common/images_test.go (hunk 2 above)
Comment (mattmoor) [inline pkg/reconciler/knativeeventing/common/images_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/reconciler_test.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2018 The Knative Authors
```
Comment (mattmoor) [inline pkg/reconciler/reconciler_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/knativeeventing/common/extensions_test.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+func fakePlatform(kubeClient kubernetes.Interface, logger *zap.SugaredLogger) (mf.Transformer, error) {
+	return fakeTransformer(), nil
+}
+
+func fakeTransformer() mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		return nil
+	}
+}
```
Comment (mattmoor) [inline pkg/reconciler/knativeeventing/common/extensions_test.go]: Add trailing newline:
```suggestion
}

```

**Comment by aliok** [comment]
Hmm, strange. My /lgtm shouldn't cause merge I think since I am not in the OWNERS file.

cc @k4leung4

**Pull request merged**